### PR TITLE
[wasm] Build improvements based on feedback

### PR DESCRIPTION
**Pull request by radical**
- Fix for a incremental build issue
- Change native build defaults for `Debug` config, to make it faster (thanks @SteveSandersonMS, @mattleibow)

- MonoAOTCompiler:
  - Skip unmanaged assemblies, and emit a warning
- PInvokeTableGenerator:
  - Fix to not fail in presence of native variadic functions, and pinvokes with function pointers (eg. used by sqlite)

### CODE REVIEW

File: src/mono/wasm/build/WasmApp.Native.targets [hunk 1]
```diff
@@ -156,6 +157,7 @@
 
       <_EmccAssertionLevelDefault>0</_EmccAssertionLevelDefault>
       <_EmccOptimizationFlagDefault Condition="'$(_WasmDevel)' == 'true'">-O0 -s ASSERTIONS=$(_EmccAssertionLevelDefault)</_EmccOptimizationFlagDefault>
+      <_EmccOptimizationFlagDefault Condition="'$(_EmccOptimizationFlagDefault)' == '' and '$(Configuration)' == 'Debug'">-O1</_EmccOptimizationFlagDefault>
       <_EmccOptimizationFlagDefault Condition="'$(_EmccOptimizationFlagDefault)' == '' and '$(OS)' != 'Windows_NT' and '$(Configuration)' == 'Debug'">-Os</_EmccOptimizationFlagDefault>
```
[the remark this answers is not in the document]
Comment (lewing) [reply src/mono/wasm/build/WasmApp.Native.targets:161]: The next line looks suspect too.

File: src/mono/wasm/build/WasmApp.Native.targets [hunk 2]
```diff
@@ -156,6 +157,7 @@
...
       <_EmccOptimizationFlagDefault Condition="'$(_EmccOptimizationFlagDefault)' == '' and '$(Configuration)' != 'Debug'">-Oz</_EmccOptimizationFlagDefault>
```
Comment (radical) [inline src/mono/wasm/build/WasmApp.Native.targets:162]:
```suggestion
```

**Review by steveisok — APPROVED** [review]

**Comment by radical** [comment]
wasm test failure is https://github.com/dotnet/runtime/issues/48717

**Comment by radical** [comment]
Coreclr failures are https://github.com/dotnet/runtime/issues/10680 , unrelated to this PR.

**Comment by radical** [comment]
Android failure is https://github.com/dotnet/runtime/issues/59779 .

**Review by lewing — APPROVED** [review]

**Pull request merged**